Add dependsOn explanation (#253)
weaveworks/profiles: userdocs/profiles.dev/docs/installer-docs/generate-local-manifests.md

  --out relative-path \
  <catalog name>/<profile>
```

Consider the following installation:

```
pctl install --git-repository flux-system/flux-system nginx-catalog/nginx/v2.0.1
generating profile installation from source: catalog entry nginx-catalog/nginx/v2.0.1
installation completed successfully
```

Let's take a look inside:

```
tree nginx
nginx
├── artifacts
│   └── bitnami-nginx
│       ├── helm-chart
│       │   ├── ConfigMap.yaml
│       │   ├── HelmRelease.yaml
│       │   └── HelmRepository.yaml
│       ├── kustomization.yaml
│       └── kustomize-flux.yaml
└── profile-installation.yaml
```

> This profile installs an `nginx` using a remote helm chart. The folders contain the following files in order:

Reviewer: `an nginx` what?

Author: Uh... application? package? software? thing? :D

Reviewer: I think either:

> This profile installs `nginx`

or

> This profile installs an `nginx` <insert descriptive thing>

"an nginx" is weird sounding

Author: Fair point. I'll think of something. :)

* bitnami-nginx - the name of the artifact
* helm-chart - contains the resources which install the actual nginx

> * ConfigMap.yaml - contains any settings which the artifact had

Reviewer: ```suggestion
* ConfigMap.yaml - contains any default values set by the author on the artifact
```

* HelmRelease.yaml - contains the HelmRelease object which installs nginx

> * HelmRepository.yaml - contains the definition of the helm chart repository where the chart is located at

Reviewer: ```suggestion
* HelmRepository.yaml - contains the definition of the helm chart repository where the chart is located
```

> * kustomization.yaml - this is a file which tells flux what to install -- more on this later

Reviewer: "more on this later" -> where is this more? is it the bit at the bottom? maybe "see below" is easier?

Author: oh yes. I wanted to fix this reference. cheers.

* kustomize-flux.yaml - this is a Kustomization object which deals with [dependencies](/docs/author-docs/dependencies)
* profiles-installation.yaml - contains information about the profile -- mainly used by pctl

What is `kustomization.yaml`? This is to prevent flux installing whatever there is in the `helm-chart` folder. The `helm-chart`

> folder can contain local resources such as, helm chart definition, READMEs and non-kubernetes objects. If flux would try

Reviewer: ```suggestion
folder can contain local resources such as helm chart definitions, READMEs and non-kubernetes objects. If flux were to try to
```

> installing those, it would fail.

Reviewer: ```suggestion
instal those, it would fail.
```

`kustomization.yaml` contains a single resource line:

```yaml
resources:
- kustomize-flux.yaml
```

> Which means, flux will only install the resource defined in this file. The Kustomization object in `kustomize-flux.yaml`

Reviewer: ```suggestion
Which means flux will only install the resource defined in this file. The Kustomization object in `kustomize-flux.yaml`
```
